src/rust/ide/lib/span-tree/src/node.rs

```diff
@@ -48,8 +60,45 @@ impl Kind {
             _              => false
         }
     }
+
+    /// Match the value with `Kind::Operation{..}`.
+    pub fn is_operation(self) -> bool {
+        match self {
+            Self::Operation => true,
+            _               => false
+        }
+    }
+
+    /// Match the value with `Kind::Empty{..}` but not `Kind::Empty(ExpectedArgument(_))`.
+    pub fn is_positional_insertion_point(self) -> bool {
+        match self {
+            Self::Empty(InsertType::ExpectedArgument(_)) => false,
+            Self::Empty(_)                               => true,
+            _                                            => false
+        }
+    }
+
+    /// Match the value with `Kind::Empty(ExpectedArgument(_))`.
+    pub fn is_expected_argument(self) -> bool {
+        match self {
+            Self::Empty(InsertType::ExpectedArgument(_)) => true,
+            _                                            => false
+        }
+    }
 }
 
+impl Default for Kind {
+    fn default() -> Self {
+        Self::Empty(default())
```

Comment:
`Empty` is not a good default for kind; empty nodes (and their insertion point) have sense only in context of their parent (so we may realize what kind of insertion it is). If default node does not have any parent, it should be "root" instead. But I don't see point in creating sole root node - instead you can just use `SpanTree::default()`

Reply:
As we agreed on the phone, lets leave it, as explained above